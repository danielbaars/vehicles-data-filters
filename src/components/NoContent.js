import React from 'react';
import PropTypes from 'prop-types';

import Outer from '../grid/Outer';
import Column from '../grid/Column';
import FramedBox from './FramedBox';

const NoContent = ({ fetchData }) => {
  return (
    <Outer className='no-content'>
      <Column>
        <FramedBox className='no-content'>
          <h4 className='no-content__text'>Sorry! Our request for the data timed out...</h4>
          <button onClick={() => fetchData()} className='button no-content__button'>Try again</button>
        </FramedBox>
      </Column>
    </Outer>
  );
};

export default NoContent;


NoContent.propTypes = {
  fetchData: PropTypes.func.isRequired,
};
