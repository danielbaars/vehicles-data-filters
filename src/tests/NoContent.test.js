import React from 'react';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';

import NoContent from '../components/NoContent';

describe('NoContent', () => {
  const mockFetchData = jest.fn();
  const component = <NoContent fetchData={mockFetchData} />;
  const sComponent = shallow(component);
  it('renders and matches our snapshot', () => {
    const rComponent = renderer.create(component);
    const tree = rComponent.toJSON();
    expect(tree).toMatchSnapshot();
  });
  it('calls the passed in fetchData function when the try again button is clicked', () => {
    sComponent.find('.no-content__button').simulate('click');
    expect(mockFetchData).toBeCalled();
  });
});
