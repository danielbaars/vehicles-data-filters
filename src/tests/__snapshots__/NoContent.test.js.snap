// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NoContent renders and matches our snapshot 1`] = `
<div
  className="no-content__outer outer"
>
  <div
    className="grid-container"
  >
    <div
      className="grid-x grid-margin-x "
    >
      <div
        className="small-12 cell"
      >
        <div
          className="no-content inner"
        >
          <div
            className="no-content__box box"
          >
            <h4
              className="no-content__text"
            >
              Sorry! Our request for the data timed out...
            </h4>
            <button
              className="button no-content__button"
              onClick={[Function]}
            >
              Try again
            </button>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
